extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/util/SqlStatementResourceHelper.java

```diff
@@ -361,22 +364,42 @@ public static MSQStagesReport.Stage getFinalStage(MSQTaskReportPayload msqTaskRe
     return null;
   }
 
-  public static Map<String, Object> getQueryExceptionDetails(Map<String, Object> payload)
+  @Nullable
+  private static MSQErrorReport getQueryExceptionDetails(MSQTaskReportPayload payload)
   {
-    return getMap(getMap(payload, "status"), "errorReport");
+    return payload == null ? null : payload.getStatus().getErrorReport();
   }
 
-  public static Map<String, Object> getMap(Map<String, Object> map, String key)
+  @Nullable
+  public static MSQTaskReportPayload getPayload(TaskReport.ReportMap reportMap)
   {
-    if (map == null) {
+    if (reportMap == null) {
       return null;
     }
-    return (Map<String, Object>) map.get(key);
+
+    com.google.common.base.Optional<MSQTaskReport> report = reportMap.findReport("multiStageQuery");
```

Comment:
Just an observation: it's slightly weird that we have `Optional`  usages from both `guava` and `java.util` packages in the same class.

Reply:
Fixed.